examples/asr/speech_to_text.py

```diff
@@ -62,16 +55,86 @@
+        model.train_ds.manifest_filepath="<path to manifest file>" \
+        model.validation_ds.manifest_filepath="<path to manifest file>" \
+        trainer.gpus=-1 \
+        trainer.max_epochs=50 \
+        +init_from_nemo_model="<path to .nemo model file>"
+```
+
+2) Finetune from a pretrained model (via NGC)
+
+```sh
+    python examples/asr/speech_to_text.py \
+        --config-path=<path to dir of configs> \
+        --config-name=<name of config without .yaml>) \
+        model.train_ds.manifest_filepath="<path to manifest file>" \
+        model.validation_ds.manifest_filepath="<path to manifest file>" \
+        trainer.gpus=-1 \
+        trainer.max_epochs=50 \
+        +init_from_pretrained_model="<name of pretrained checkpoint>"        
+```
 """
 
+import pytorch_lightning as pl
+from omegaconf import OmegaConf, open_dict
+
+from nemo.collections.asr.models import EncDecCTCModel
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager
+
+
+def restore_weights_if_required(model: EncDecCTCModel, cfg: OmegaConf):
+    if 'init_from_nemo_model' not in cfg and 'init_from_pretrained_model' not in cfg:
+        # model weights do not need to be restored
+        return
+
+    if 'init_from_nemo_model' in cfg and 'init_from_pretrained_model' in cfg:
+        raise ValueError("Cannot pass both `init_from_nemo_model` and `init_from_pretrained_model` to config!")
+
+    if 'init_from_nemo_model' in cfg and cfg.init_from_nemo_model is not None:
+        with open_dict(cfg):
+            # Restore model
+            model_path = cfg.pop('init_from_nemo_model')
+            restored_model = EncDecCTCModel.restore_from(model_path, map_location='cpu', strict=True)
+
+            # Restore checkpoint into current model
+            model.load_state_dict(restored_model.state_dict(), strict=False)
+            logging.info(f'Model checkpoint restored from nemo file with path : `{model_path}`')
+
+            del restored_model
+
+    if 'init_from_pretrained_model' in cfg and cfg.init_from_pretrained_model is not None:
+        with open_dict(cfg):
+            # Restore model
+            model_name = cfg.pop('init_from_pretrained_model')
+            restored_model = EncDecCTCModel.from_pretrained(model_name, map_location='cpu', strict=True)
+
+            # Restore checkpoint into current model
+            model.load_state_dict(restored_model.state_dict(), strict=False)
+            logging.info(f'Model checkpoint restored from pretrained chackpoint with name : `{model_name}`')
+
+            del restored_model
+
 
 @hydra_runner(config_path="conf", config_name="config")
 def main(cfg):
     logging.info(f'Hydra config: {OmegaConf.to_yaml(cfg)}')
+
     trainer = pl.Trainer(**cfg.trainer)
     exp_manager(trainer, cfg.get("exp_manager", None))
     asr_model = EncDecCTCModel(cfg=cfg.model, trainer=trainer)
 
+    restore_weights_if_required(asr_model, cfg)
```

Comment:
The word restore conveys the meaning that this model has some weights, it lost them and we want them back. Init may be a more appropriate word here.